tests/test_data/test_datasets/test_custom_dataset.py

```diff
@@ -23,61 +99,94 @@ def test_custom_classes_override_default(dataset):
 
     original_classes = dataset_class.CLASSES
 
+    # create dummy data
+    tmp_dir = tempfile.TemporaryDirectory()
+    if dataset in ['CocoDataset', 'CityscapesDataset']:
+        fake_file = osp.join(tmp_dir.name, 'fake_data.json')
+        _create_dummy_coco_json(fake_file)
+    elif dataset in ['VOCDataset']:
+        fake_file = 'tests/data/VOCdevkit/VOC2007/ImageSets/Main/trainval.txt'
+    else:
+        fake_file = osp.join(tmp_dir.name, 'fake_data.pkl')
+        _create_dummy_custom_pkl(fake_file)
+
     # Test setting classes as a tuple
     custom_dataset = dataset_class(
-        ann_file=MagicMock(),
+        ann_file=fake_file,
         pipeline=[],
         classes=('bus', 'car'),
         test_mode=True,
-        img_prefix='VOC2007' if dataset == 'VOCDataset' else '')
+        img_prefix='./tests/data/VOCdevkit/VOC2007/' if dataset == 'VOCDataset' else '')
```

Comment:
why not ` img_prefix='VOC2007' if dataset == 'VOCDataset' else ''`

Reply:
Because I need to read the label files in the folder path `tests/data/VOCdevkit/VOC2007/` to count the instance number.
`img_prefix='VOC2007' if dataset == 'VOCDataset' else ''` does not refer to the correct path